src/main/java/com/onegini/examples/resourcegateway/model/TokenIntrospectionResult.java

```diff
@@ -15,107 +20,12 @@
   private int usageLimit;
   private TokenType[] amr;
 
-  public boolean isActive() {
-    return active;
-  }
-
-  public void setActive(final boolean active) {
-    this.active = active;
-  }
-
-  public String getScope() {
-    return scope;
-  }
-
-  public void setScope(final String scope) {
-    this.scope = scope;
-  }
-
-  public String getClientId() {
-    return clientId;
-  }
-
-  public void setClientId(final String clientId) {
-    this.clientId = clientId;
-  }
-
-  public String getTokenType() {
-    return tokenType;
-  }
-
-  public void setTokenType(final String tokenType) {
-    this.tokenType = tokenType;
-  }
-
-  public int getExp() {
-    return exp;
-  }
-
-  public void setExp(final int exp) {
-    this.exp = exp;
-  }
-
-  public int getIat() {
-    return iat;
-  }
-
-  public void setIat(final int iat) {
-    this.iat = iat;
-  }
-
-  public int getNbf() {
-    return nbf;
-  }
-
-  public void setNbf(final int nbf) {
-    this.nbf = nbf;
-  }
-
-  public String getSub() {
-    return sub;
-  }
-
-  public void setSub(final String sub) {
-    this.sub = sub;
-  }
-
-  public int getUsageLimit() {
-    return usageLimit;
-  }
-
-  public void setUsageLimit(final int usageLimit) {
-    this.usageLimit = usageLimit;
-  }
-
   public TokenType[] getAmr() {
-    return amr;
+    return ArrayUtils.clone(amr);
   }
 
   public void setAmr(final TokenType[] amr) {
-    this.amr = amr;
-  }
-
-  public String getAppIdentifier() {
-    return appIdentifier;
-  }
-
-  public void setAppIdentifier(final String appIdentifier) {
-    this.appIdentifier = appIdentifier;
-  }
-
-  public String getAppVersion() {
-    return appVersion;
-  }
-
-  public void setAppVersion(final String appVersion) {
-    this.appVersion = appVersion;
+    this.amr = ArrayUtils.clone(amr);
```

Comment:
Wonder why do we need to do a shallow copy of the array here? 🤔  Wouldn't it be easier to expect a collection instead?

Reply:
I have no clue, but I have changed the field to be a `Collection<TokenType>` which removes the need to shallow copy it.